daml-lf/governance.rst

```diff
@@ -5,109 +5,144 @@ DAML-LF Governance process
-  replay every transaction ever run on the ledger. However the ledger is
-  free to refuse to accept new code running with older DAML-LF versions
-  while still maintaining support for existing code internally.
-
-* Client libraries and other tooling mostly care about DAML-LF values
-  and DAML-LF serializable types. Deciding how these components handle
-  versions is not part of the DAML-LF governance progress. For
-  example, a certain version of the DAML-LF Scala codegen might
-  support a specified set of DAML-LF versions -- much like ledger
-  implementations.
-
-Values
-------
+  labels "component/daml-lf" and "discussion".
+
+Language
+--------
+
+Some version bumps might not change the ``.proto`` at all, but just
+add / change semantics in the specification. For example, the string
+format for decimal numbers in DAML-LF values might be made more
+lenient, resulting in a version bump with no ``.proto`` schema
+changes (but probably a comment).
+
+Also note that the DAML-LF versioning is independent from the DAML
+surface language versioning.
+
+"dev" version
+~~~~~~~~~~~~~
+
+Every DAML-LF major version includes a minor version, the *dev* version,
+which we use as staging area for the next stable minor version of that
+major version.
+
+The "dev" version can be changed freely without
+compatibility considerations to prior "dev" versions.  Since "dev" is
+always considered to be newer than every stable minor version of that
+major version, it must be backward-compatible with all such stable
+versions.
+
+All newly-implemented minor version features or changes must be staged
+in the *dev* version for later release as a stable group of
+backward-compatible features.
+
+The DAML-LF dev version is enabled in the sandbox and ledger server, but
+will never be emitted by damlc unless explicitly requested via
+``--target 1.dev`` or similar.
 
-Each DAML-LF value and DAML-LF transaction comes with a version number,
-indicating how the respective specification should be consulted for
-rules regarding that value or transaction.
-
-Every change to the ``.proto`` specification must yield a specification
-capable of consuming all values produced by any prior version of the
-specification. New fields can be added, the human-readable names of
-existing fields can be changed, and the semantics of fields can be
-changed according to the version number included with the enclosing
-value or transaction.  Moreover, messages produced via the new proto may
-be consumed via any prior proto, but *only* to the extent of completing
-without error and providing the version number.  The remaining contents
-of the message are neither required to make sense, nor preserve the
-semantics of older versions.
+Associated Transaction version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Except for `1.6`, `1.7`, and `1.8` each language version is associated
+to an unique transaction version, called the associated transaction
```

Comment:
Aren't each of them assigned to a unique version as well, although that version is the same for all of them? The non-uniqueness only exists the other way around, in the sense that this version is not associated to a unique language version.

Reply:
fix.